packages/plugins/ui-theme/src/store/models/manifest.ts

```diff
@@ -0,0 +1,93 @@
+import { createModel } from '@rematch/core';
+import { Package } from '@verdaccio/types';
+import { PackageMetaInterface } from 'types/packageMeta';
+
+import API from '../../providers/API/api';
+
+import type { RootModel } from '.';
+
+function isPackageVersionValid(
+  packageMeta: Partial<PackageMetaInterface>,
+  packageVersion?: string
+): boolean {
+  if (!packageVersion || typeof packageVersion === 'undefined') {
+    // if is undefined, that means versions does not exist, we continue
+    return true;
+  }
+
+  if (packageMeta.versions) {
+    return Object.keys(packageMeta.versions).includes(packageVersion);
+  }
+
+  return false;
+}
+
+export const manifest = createModel<RootModel>()({
+  state: {},
+  reducers: {
+    notFound(state) {
+      return {
+        ...state,
+        hasNotBeenFound: true,
+        manifest: undefined,
+        packageName: undefined,
+        packageVersion: undefined,
+        readme: undefined,
+      };
+    },
+    clearError(state) {
+      return {
+        ...state,
+        isError: null,
+      };
+    },
+    isError(state) {
+      return {
+        ...state,
+        isError: true,
+        hasNotBeenFound: false,
+        manifest: undefined,
+        packageName: undefined,
+        packageVersion: undefined,
+        readme: undefined,
+      };
+    },
+    saveManifest(state, { packageName, packageVersion, manifest, readme }) {
+      return {
+        ...state,
+        manifest,
+        packageName,
+        packageVersion,
+        readme,
+        hasNotBeenFound: false,
+      };
+    },
+  },
+  effects: (dispatch) => ({
+    async getManifest({ packageName, packageVersion }, state) {
+      const basePath = state.configuration.config.base;
+      try {
+        if (!isPackageVersionValid(packageName, packageVersion)) {
+          throw new Error('not found');
+        }
+
+        const manifest: Package = await API.request(
+          `${basePath}-/verdaccio/sidebar/${packageName}${
+            packageVersion ? `?v=${packageVersion}` : ''
+          }`
+        );
+        const readme: string = await API.request<string>(
+          `${basePath}-/verdaccio/package/readme/${packageName}${
+            packageVersion ? `?v=${packageVersion}` : ''
+          }`,
+          'GET'
+        );
+        // @ts-ignore
+        dispatch.manifest.saveManifest({ packageName, packageVersion, manifest, readme });
```

Comment:
@semoal do you have an idea why I do have this error `Expected 0 arguments, but got 1` if I remove the ts-ignore?

Reply:
Type the payload on the effect param, that should work. We don't have any way of inferring the payload from each effect. Our typing system is as complex as we reached some design limitations of typescript 🤟🏻

Reply:
The code itself works fine, if I understood correctly is not possible to apply any typing, right? I've tried few things but no success.

Reply:
No no, it's completely possible. I'll take a look this morning